### PR TITLE
Better resize methods

### PR DESCRIPTION
**Pull request by jakearchibald**

### CODE REVIEW

File: codecs/resize/pkg/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# 🦀🕸️ `wasm-pack-template`
```
Comment (surma) [inline codecs/resize/pkg/README.md]: I should `.gitignore` this one

File: package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "start": "webpack-dev-server --host 0.0.0.0 --hot",
     "build": "webpack -p",
-    "lint": "tslint -c tslint.json  -p tsconfig.json -t verbose 'src/**/*.{ts,tsx,js,jsx}'",
```
Comment (surma) [inline package.json]: Is this change intentional?

File: src/codecs/resize/processor.ts [hunk 3]
```diff
@@ -1,49 +1,50 @@
...
-    ({ sx, sy, sw, sh } = getContainOffsets(sw, sh, opts.width, opts.height));
+    const { sx, sy, sw, sh } = getContainOffsets(data.width, data.height, opts.width, opts.height);
+    input = crop(input, Math.round(sx), Math.round(sy), Math.round(sw), Math.round(sh));
   }
 
-  return drawableToImageData(data, {
-    sx, sy, sw, sh,
-    width: opts.width, height: opts.height,
-  });
+  await wasm_bindgen(wasmUrl);
```
Comment (surma) [inline src/codecs/resize/processor.ts]: Does this cause a fetch/initialization every time? Could we keep the instance around?

File: package.json (hunk 2 above)
Comment (jakearchibald) [reply package.json]: Yeah, this is what was causing it to lint all the `.d.ts` files. It still lints regular files.

File: src/codecs/resize/processor.ts (hunk 3 above)
Comment (jakearchibald) [reply src/codecs/resize/processor.ts]: Good call.

**Comment by jakearchibald** [comment]
Cached the module. I'll have a click around in a few browsers & ship it. Yay for achieving something today!
